File: Objective-C/CustomAlertWithSwift/CustomAlertWithSwift/ViewController.m
//
//  ViewController.m
//  CustomAlertWithSwift
//
//  Created by Nick on 2023/06/27.
//

#import "ViewController.h"
#import "CustomAlertWithSwift-Swift.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)alertButtonTapped:(UIButton *)sender {
    CustomAlertView * customAlertView = [[CustomAlertView alloc] initWithTitle: @"제목입니다" content: @"내용입니다" confirmText: @"확인" cancelText:@"취소"];
    customAlertView.modalPresentationStyle = UIModalPresentationOverFullScreen;
    [self presentViewController:customAlertView animated:NO completion:nil];
}


@end

File: UI Practice/ExpandableView/ExpandableView/ViewController.swift
//
//  ViewController.swift
//  ExpandableView
//
//  Created by Nick on 2023/10/17.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var upperView: UIStackView!
    @IBOutlet weak var buttonList: UIStackView!
    var isExpanded: Bool = false
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let tap = UITapGestureRecognizer(target: self, action: #selector(upperViewTapped))
        upperView.addGestureRecognizer(tap)
    }
    
    @objc
    func upperViewTapped(_ sender: UITapGestureRecognizer) {
        print("Tapped")
        UIView.animate(withDuration: 0.2) {
            self.buttonList.isHidden = !self.isExpanded
            self.view.layoutIfNeeded()
       }
        isExpanded.toggle()
        
        print("remove")
    }
    
}

